### ISSUE TITLE
Test spec succeeding locally, failing during CI

### ISSUE THREAD

**Issue by abbeycode** [CocoaPods/CocoaPods#7643]
* [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Run a script that includes `pod lib lint --verbose | sed -l '/xctest\[/d; /^$/d'`, for pod that contains a test spec, locally in Terminal, and in Travis CI's latest Xcode environment

## What did you expect to happen?

Test spec validation either succeeds or fails on my local machine and on the CI server

## What happened instead?

Tests fail during CI, but succeed locally. I see these errors from the CI logs:

>    Testing failed:
>    	Test target UnzipKit-Unit-Tests encountered an error (Early unexpected exit, operation never finished bootstrapping - no restart will be attempted)
>    ** TEST FAILED **

And then a little further down:

>     Usage: xctest [-XCTest All | <TestCaseClassName/testMethodName>] <path of unit to be tested>
>    
>    Failure: No test configuration provided.

#### Stack

```
   CocoaPods : 1.5.0
        Ruby : ruby 2.3.3p222 (2016-11-21 revision 56859) [universal.x86_64-darwin17]
    RubyGems : 2.5.2
        Host : Mac OS X 10.13.4 (17E199)
       Xcode : 9.3 (9E145)
         Git : git version 2.15.1 (Apple Git-101)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 80423e252159d07bef88111a12accf1407e34370
```

#### Installation Source

```
Executable Path: /Users/dfrankel/.gem/ruby/2.3.0/bin/pod
```

#### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

#### Stack

```
   CocoaPods : 1.5.0
        Ruby : ruby 2.4.2p198 (2017-09-14 revision 59899) [x86_64-darwin16]
    RubyGems : 2.6.13
        Host : Mac OS X 10.13.3 (17D47)
       Xcode : 9.3 (9E145)
         Git : git version 2.16.1
Ruby lib dir : /Users/travis/.rvm/rubies/ruby-2.4.2/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 869b916bd7cff20062e6948cca3bbf2b296421f5
```

#### Installation Source

```
Executable Path: /Users/travis/.rvm/gems/ruby-2.4.2/bin/pod
```

#### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

## Project that demonstrates the issue

[The `testspec` branch of UnzipKit](https://github.com/abbeycode/UnzipKit/tree/testspec)

**Comment by DanToml** [comment]
Please can you try using a Gemfile to ensure the same versions of everything on CI vs locally? There's a good chance this is caused by a version mismatch or simulator issues on travis.

**Comment by abbeycode** [comment]
@DanToml I'm on the same versions of all the CocoaPods gems (as demonstrated with the `pod env` output). Are there other gems I should specify in a Gemfile?

**Comment by dnkoutso** [comment]
@abbeycode are you sure this is a cocoapods library issue? Seems like a CI configuration issue to me?

**Comment by abbeycode** [comment]
@dnkoutso I'm not sure, but I'll give you the reasons I suspect so. When I run my unit tests directly (with my project's scheme), they succeed. Also, no particular unit test is failing, but the test step causes an unexpected early exit happen to happen, implying something about the way the testing is performed. Take a look at [this build](https://travis-ci.org/abbeycode/UnzipKit/builds/367249654), which runs my Mac and iOS unit tests in the project's own scheme, as well as the `pod validate` command. Only the CocoaPods validation fails.

**Comment by dnkoutso** [comment]
have you tried `lint` locally and the tests pass?

**Comment by abbeycode** [comment]
@dnkoutso Yes, the tests run and pass locally. That's part of what puzzles me

**Comment by dnkoutso** [comment]
@abbeycode I assume you mean it passes locally using `pod lib lint`...

Is there a chance CI is using `pod spec lint` and the tag of your library is not updated?

I am inclined to close this since it appears that locally the test spec passes which makes me believe this is a CI configuration issue...would love to know the root cause though.

**Comment by segiddins** [comment]
> (Early unexpected exit, operation never finished bootstrapping - no restart will be attempted)

is it possible an iOS simulator failed to launch on CI?

**Comment by abbeycode** [comment]
@dnkoutso I'm using `pod lib lint` in both places, called from the same script.

@segiddins Is there any way to affirmatively tell from the logs if a simulator crashed?

Another reason I'm disinclined to blame the CI environment is that this was happening with the Xcode 9.2/macOS 10.12 image, and continued after I upgraded to the 9.3/10.13 image.

Would either of you mind trying to run the lint yourselves and see if it works, as a third data point? Clone the `testspec` branch of UnzipKit (https://github.com/abbeycode/UnzipKit.git) and run `Scripts/cocoapod-validate.sh`. That should be all that's required.

**Comment by abbeycode** [comment]
I did more searching on the error message "Failure: No test configuration provided" and found #7608. I was able to fix it by updating my podspec target iOS 9.0 instead of 8.0. I'm not totally sure why it was succeeding locally - maybe Travis keep the old SDK around? Either way, I'm all set. Thanks for the help!
